lib/router/handler-info/unresolved-handler-info-by-object.js

```diff
@@ -25,7 +25,8 @@ var UnresolvedHandlerInfoByObject = subclass(HandlerInfo, {
   serialize: function(_model) {
     var model = _model || this.context,
         names = this.names,
-        serializer = this.serializer || (this.handler && this.handler.serialize);
+        // this.serializer is intentionally unbound since getSerializer is expected to return a stateless function
+        serializer = this.serializer || (this.handler && this.handler.serialize && this.handler.serialize.bind(this.handler));
```

Comment:
I would prefer to refactor this a bit to avoid needing to `.bind` again for every call to the same handler's `serialize`.  I think the simplest way woul dbe to remove this `serializer` local variable, and update the `if` conditional below to:

```js
if (this.serializer) {
  // invoke this.serializer unbound (getSerializer returns a stateless function)
  return (0, this.serializer)(model, names);
} else if (this.handler && this.handler.serialize) {
  return this.handler.serialize(model, names);
}
```